### ISSUE TITLE
Implement sampling profiler for both Lua and C

### ISSUE THREAD

**Issue by rosik** [tarantool/tarantool#4001]
This is a concept. I think we have a chance to enhance `jit.p` profiler and make it capture both C ald Lua stacks simultaneously.

How it works now:

```lua
-- script-1.lua
local jitp = require('jit.p')
jitp.start('3')

local function highload()
	for i = 1, 20 * 1000 * 1000 do
		debug.getlocal(1, 1)
	end
end
local function foo()
	highload()
end

highload()
foo()

jitp.stop()
```

```console
$ tarantool ./script-1.lua 
50%  highload < foo < test.lua:0
50%  highload < test.lua:0
```

The implementation of `jit.p` is here: https://github.com/tarantool/luajit/blob/tarantool-1.7/src/jit/p.lua

And the most interesting part: https://github.com/tarantool/luajit/blob/tarantool-1.7/src/lj_profile.c#L356 - function `const char *luaJIT_profile_dumpstack(lua_State *L, const char *fmt, int depth, size_t *len)`.

Jit.p calls it periodically to capture stack. If we substitute it with our own implementation, we can gather stack traces from both C and Lua, thanks to #3538.

Proof of concept:
```lua
-- script-2.lua
local fiber = require('fiber')
local jitp = require('jit.p')
local profile = require('jit.profile')

-- substitute dumpstack with my own implementation
function profile.dumpstack(...)
	return 'abcd:1 < efgh:2'
end
jitp.start('3')

local function highload()
	for i = 1, 20 * 1000 * 1000 do
		debug.getlocal(1, 1)
	end
end

highload()

local function foo()
	highload()
end

foo()

jitp.stop()
```

```console
$ tarantool ./script-2.lua 
100%  abcd:1 < efgh:2
```

Things to discuss:

* Is it useful?
* How can we modify fiber.info() api and trace output to produce better results?
* How much code in `lj_profile.c` do we want to rewrite?

**Comment by rosik** [comment]
Btw this seems to be related with #781

**Comment by igormunkin** [comment]
Resolved in scope of #781.
